main.tf

```
@@ -134,7 +139,7 @@ resource "aws_iam_role" "vpc_flow_logs_to_splunk_kinesis_firehose_lambda" {
   })
 
   inline_policy {
-    name = "${var.vpc_name}-vpc-flow-logs-to-splunk-kinesis-firehose-lambda"
+    name = substr("${local.name}-vpc-flow-logs-to-splunk-kinesis-firehose-lambda", 0, 64)
```

```suggestion
    name = substr("${local.name}-vpc-flow-logs-to-splunk-fh-lambda", 0, 64)
```
While substr will prevent a failure of a name which is too long, I would keep the fixed name part shorter. 
Otherwise, it can be truncated in an odd way... (e.g. you can see the resource being named `${local.name}-vpc-flow-logs-to-splunk-kines`

---

hmmm, makes sense. I'll see what I can do :)

---

a nicer way to comply with the 64 chars requirement https://github.com/app-sre/terraform-aws-vpc-flow-logs-splunk/pull/8/commits/e745005f130cb95702609c97d8b39c3a43007932

---

> a nicer way to comply with the 64 chars requirement [e745005](https://github.com/app-sre/terraform-aws-vpc-flow-logs-splunk/commit/e745005f130cb95702609c97d8b39c3a43007932)

I like it better, but still, if the VPC ID + predefined suffix exceed 64 characters, it's problem for the module consumer, as they can't make the VPC ID shorter (and maybe not even the VPC name) and they can't make the suffix fix string shorter either. 

Don't forget that the reason I had to change the module is because I got an error about exceeding the maximum number of characters allowed for a resource.

Maybe you can allow the module consumer the option to provide an alternative suffix?
Or just go with something shorter to lower the chances of exceeding 64 characters. 
WDYT?

---

hmmm, the vpc_id is 21 characters long and that's not likely to change soon... https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/resource-ids.html

With a 21 characters VPC name it works fine, e.g.

```
vpc-036cb989c3d8c781c-vpc-flow-logs-to-splunk-kinesis-firehose
```

This has 62 characters.

I can add a validation to the vpc_id length too just to make sure this is not going blow.

---

Alright. Sounds good to me.

---

added validation in ef97426